test/e2e/tests/helper.py

```diff
@@ -0,0 +1,100 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Helper functions for Lambda e2e tests
+"""
+
+import logging
+
+class LambdaValidator:
+    def __init__(self, lambda_client):
+        self.lambda_client = lambda_client
+
+    def get_function(self, function_name: str) -> dict:
+        try:
+            resp = self.lambda_client.get_function(
+                FunctionName=function_name
+            )
+            return resp
+
+        except Exception as e:
+            logging.debug(e)
+            return None
+
+    def get_function_concurrency(self, function_name: str) -> int:
+        try:
+            resp = self.lambda_client.get_function_concurrency(
+                FunctionName=function_name
+            )
+            return resp['ReservedConcurrentExecutions']
+
+        except Exception as e:
+            logging.debug(e)
+            return None
+
+    def get_function_code_signing_config(self, function_name: str) -> int:
+        try:
+            resp = self.lambda_client.get_function_code_signing_config(
+                FunctionName=function_name
+            )
+            return resp['CodeSigningConfigArn']
+
+        except Exception as e:
+            logging.debug(e)
+            return None
+
+    def assert_function_exists(self, function_name: str, exists=True):
+        assert (self.get_function(function_name) is not None) == exists
```

Comment:
Here and below, I think I've stated this before, but I think the `exists=True` kwarg thing is an anti-pattern and should be avoided.

Instead, I believe `assert` should always be placed in the test code itself and helper methods should take an action, not perform assertions.

So, this should instead be:

```python
def function_exists(self, function_name):
    self.get_function(function_name) is not None
```

and calling code should assert existence (or not) like so:

```python
    assert helper.function_exists(func_name)
    assert not helper.function_exists(func_name)
```

Reply:
i'm not super familiar with python best practices regarding args and kwargs, thank you! Will make the change accordingly